ProjectManagement.BusinessLogic/Services/Implementation/FileService.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using nClam;
+using ProjectManagement.BusinessLogic.Exceptions;
+using ProjectManagement.BusinessLogic.Services.Interfaces;
+using ProjectManagement.DataAccess.Repositories.Interfaces;
+using ProjectManagement.Domain.Models;
+using BCryptNet = BCrypt.Net.BCrypt;
+
+namespace ProjectManagement.BusinessLogic.Services.Implementation
+{
+    public class FileService : IFileService
+    {
+        private readonly ILogger<FileService> _logger;
+        private readonly IConfiguration _configuration;
+   
+        public FileService(ILogger<FileService> logger, IConfiguration configuration)
+        {
+            _logger = logger;
+            _configuration = configuration;
+        }
+        
+        public async Task ScanFileForVirusesAsync(IFormFile file)
+        {
+            bool? isFileInfected = await IsFileInfectedAsync(file);
+            if(isFileInfected==null)
+            {
+                string message = $"The file may be infected with a virus.";
+                this._logger.LogError(message);
+                throw new WebAppException((int)HttpStatusCode.BadGateway, message);
+            }
+            else if (isFileInfected == true)
+            {
+                string message = $"The file infected with a virus.";
+                this._logger.LogError(message);
+                throw new WebAppException((int)HttpStatusCode.BadGateway, message);
+            }
+        }
+
+        public async Task<bool?> IsFileInfectedAsync(IFormFile file)
+        {
+            var ms = new MemoryStream();
+            file.OpenReadStream().CopyTo(ms);
+            byte[] fileBytes = ms.ToArray();
+            try
+            {
+                this._logger.LogInformation("ClamAV scan begin for file {0}", file.FileName);
+                var clam = new ClamClient(this._configuration["ClamAVServer:URL"],
```

Comment:
If you want you can try to use `IOptions` for Clam configs.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-6.0